provider/core/src/helpers.rs

```diff
@@ -187,6 +187,38 @@ fn test_hash_word_32() {
     );
 }
 
+#[doc(hidden)]
+#[macro_export]
+macro_rules! gen_any_buffer_docs {
+    (ANY, $krate:path, $see_also:path) => {
+        concat!(
+            "Creates a new instance using an [`AnyProvider`](",
+            stringify!($krate),
+            "::AnyProvider).\n\n",
+            "[📚 Help choosing a constructor](",
+            stringify!($krate),
+            "::constructors)\n\n",
+            "For details on the behavior of this function, see: [`",
+            stringify!($see_also),
+            "`]"
+        )
+    };
+    (BUFFER, $krate:path, $see_also:path) => {
+        concat!(
+            "✨ **Enabled with the `\"serde\"` feature.**\n\n",
+            "Creates a new instance using a [`BufferProvider`](",
+            stringify!($krate),
+            "::BufferProvider).\n\n",
+            "[📚 Help choosing a constructor](",
```

Comment:
I'd move this under the unstable link, i.e. mention the alternative, then explain how to choose between them, not the other way around.